pkg/stub/handler.go

```diff
@@ -2,96 +2,531 @@ package stub
+	var newStatus corev1.ConditionStatus
+	if deleted {
+		err := h.secretclientwrapper.Delete("openshift", secret.Name, &metav1.DeleteOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(samplesResource, v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+		}
+		logrus.Printf("registry dockerconfig secret %s was deleted", secret.Name)
+		newStatus = corev1.ConditionFalse
+		h.registrySecret = nil
+	} else {
+		if h.registrySecret != nil {
+			currentVersion, _ := strconv.Atoi(h.registrySecret.ResourceVersion)
+			newVersion, _ := strconv.Atoi(secret.ResourceVersion)
+			if newVersion <= currentVersion {
+				return nil
+			}
+		}
+		secretToCreate := corev1.Secret{}
+		secret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+		secretToCreate.UID = ""
+
+		s, err := h.secretclientwrapper.Get("openshift", secret.Name)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(samplesResource, v1alpha1.SecretUpdateFailed, err, "failed to get registry dockerconfig secret in openshift namespace : %v")
+		}
 		if err != nil {
-			logrus.Errorf("failed to get image client : %v", err)
-			return err
+			s = nil
+		}
+		if s != nil {
+			logrus.Printf("updating dockerconfig secret %s in openshift namespace", v1alpha1.SamplesRegistryCredentials)
+			_, err = h.secretclientwrapper.Update("openshift", &secretToCreate)
+		} else {
+			logrus.Printf("creating dockerconfig secret %s in openshift namespace", v1alpha1.SamplesRegistryCredentials)
+			_, err = h.secretclientwrapper.Create("openshift", &secretToCreate)
+		}
+		if err != nil {
+			return h.processError(samplesResource, v1alpha1.SecretUpdateFailed, err, "failed to create/update registry dockerconfig secret in openshif namespace : %v")
+		}
+		newStatus = corev1.ConditionTrue
+		h.registrySecret = secret
+	}
+
+	err := h.GoodStatusUpdate(samplesResource, newStatus, v1alpha1.ImportCredentialsExist)
+	if err != nil {
+		return err
+	}
+
+	h.registrySecret = secret
+	h.samplesResource = samplesResource
+
+	return nil
+}
+
+func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if dockercfgSecret.Name != v1alpha1.SamplesRegistryCredentials {
+			return nil
 		}
-		logrus.Printf("image client %#v", imageclient)
-		templist, err := tempclient.Templates("openshift").List(metav1.ListOptions{})
-		logrus.Printf("template list %#v", templist)
-		/*err := sdk.Create(newbusyBoxPod(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("failed to create busybox pod : %v", err)
+
+		if h.samplesResource == nil {
+			if !event.Deleted {
+				h.registrySecret = dockercfgSecret
+			}
+			return nil
```

Comment:
why don't we still want to setup the secret in the openshift namespace in this scenario?

Reply:
we can't update at the status without the samplesresource ... so we either 
a) create secret now, then save data somewhere so we know to add the secret import condition when we get the sample resource
b) create the secret and add the condition at the same time

b) seemed cleaner to me, as well as less churn from our prior form

Reply:
hum. ok.